### PR TITLE
Update hugepage config in SOS Bootargs.

### PR DESCRIPTION
**Pull request by Kunhui-Li**
1. HV: refine init_vm_bootargs_info():
    1) The VM load order type condition is not needed, since the function is called only when create SOS VM or pre-launched VM;
    2) Fixed wrong parameter of fill_seed_arg() which introduced by commit 80262f060293d2ba613fa70673ebe794a511ce9c.
    3) More comments on why multiboot string could override the pre-configured VM bootargs and why append multiboot cmdline to SOS VM bootargs;
2. HV: remove SOS kernel hugepage related bootargs: 
    Hypervisor does not need to care about hugepage settings in SOS kernel, user could enable these settings in the scenario config file or GRUB menu.
3. Config_tools: config hugepage in sos kenrel cmdline:
    If there is hugepage support from board xml, config tool will add hugepagesz=1G hugepages=[size] into sos kernel cmdline, the size is calculated by memory size in G reducing 3. The reason for reducing 3 is that it is reserved for SOS VM use.

### CODE REVIEW

**Comment by wenlingz** [comment]
@terryzouhao @dongyaozu please help approve this PR, thanks

**Comment by terryzouhao** [comment]
Looks good for config tool part.

**Comment by wenlingz** [comment]
@dongyaozu please help approve this PR, thanks

File: misc/config_tools/xforms/misc_cfg.h.xsl [hunk 1]
```diff
@@ -97,7 +98,12 @@
       </xsl:otherwise>
     </xsl:choose>
   </xsl:variable>
-  <xsl:value-of select="acrn:define('SOS_BOOTARGS_DIFF', concat($quot, $bootargs, ' ', $maxcpus, ' ', $quot), '')" />
+  <xsl:variable name="hugepage">
```
Comment (gvancuts) [inline misc/config_tools/xforms/misc_cfg.h.xsl:101]: Shouldn't the name of this variable be `hugepages` (with an 's' at the end)?

Comment (Kunhui-Li) [reply misc/config_tools/xforms/misc_cfg.h.xsl:101]: Config tool will add the hugepage variable that declares the string "hugepagesz=1G hugepages=[size]" into sos kernel cmdline. The hugepages variable only mean the value of hugepages in the string "hugepagesz=1G hugepages=[size]".

Comment (gvancuts) [reply misc/config_tools/xforms/misc_cfg.h.xsl:101]: I see, thanks for the explanation. Should we use a different name, less close to the `hugepages` variable name? Perhaps something like `hugepage_kernelstring`?

**Review by gvancuts — APPROVED** [review]
Thanks!

**Comment by dongyaozu** [comment]
@JasonChenCJ

**Review by JasonChenCJ — APPROVED** [review]

**Pull request merged**